// @robotevents/client/dist/v1/index.js
import axios from 'axios';
import { CountriesClient, EventsClient, ProgramsClient, RegionsClient, SkillsClient, TeamsClient, } from './clients/index.js';
export * from './clients/index.js';
export * from './client.js';
export class RobotEventsV1Client {
    constructor(options = {}) {
        Object.defineProperty(this, "countries", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "events", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "programs", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "regions", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "skills", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "teams", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        const axiosInstance = axios.create({
            baseURL: options.url ?? 'https://robotevents.com/api',
        });
        this.countries = new CountriesClient(axiosInstance);
        this.events = new EventsClient(axiosInstance);
        this.programs = new ProgramsClient(axiosInstance);
        this.regions = new RegionsClient(axiosInstance);
        this.skills = new SkillsClient(axiosInstance);
        this.teams = new TeamsClient(axiosInstance);
    }
}
