// @robotevents/client/dist/v1/clients/team-groups.js
var _TeamGroupsRequestBuilder_programIds, _TeamGroupsRequestBuilder_seasonId, _TeamGroupsRequestBuilder_country, _TeamGroupsRequestBuilder_region;
import { Client } from '../client.js';
export class TeamGroupsClient extends Client {
    async findAll(request) {
        return this.post('/teams/latLngGrp', request?.(new TeamGroupsRequestBuilder()).build().data());
    }
}
class TeamGroupsRequestBuilder {
    constructor() {
        _TeamGroupsRequestBuilder_programIds.set(this, void 0);
        _TeamGroupsRequestBuilder_seasonId.set(this, void 0);
        _TeamGroupsRequestBuilder_country.set(this, void 0);
        _TeamGroupsRequestBuilder_region.set(this, void 0);
    }
    programIds(...value) {
        __classPrivateFieldSet(this, _TeamGroupsRequestBuilder_programIds, value, "f");
        return this;
    }
    seasonId(value) {
        __classPrivateFieldSet(this, _TeamGroupsRequestBuilder_seasonId, value, "f");
        return this;
    }
    location(country, region) {
        __classPrivateFieldSet(this, _TeamGroupsRequestBuilder_country, country, "f");
        __classPrivateFieldSet(this, _TeamGroupsRequestBuilder_region, region, "f");
        return this;
    }
    build() {
        return new TeamGroupsRequest(this);
    }
}
_TeamGroupsRequestBuilder_programIds = new WeakMap(), _TeamGroupsRequestBuilder_seasonId = new WeakMap(), _TeamGroupsRequestBuilder_country = new WeakMap(), _TeamGroupsRequestBuilder_region = new WeakMap();
Object.defineProperty(TeamGroupsRequestBuilder, "TeamGroupsRequest", {
    enumerable: true,
    configurable: true,
    writable: true,
    value: class {
        constructor(builder) {
            Object.defineProperty(this, "programIds", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "seasonId", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "country", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "region", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            this.programIds = __classPrivateFieldGet(builder, _TeamGroupsRequestBuilder_programIds, "f");
            this.seasonId = __classPrivateFieldGet(builder, _TeamGroupsRequestBuilder_seasonId, "f");
            this.country = __classPrivateFieldGet(builder, _TeamGroupsRequestBuilder_country, "f");
            this.region = __classPrivateFieldGet(builder, _TeamGroupsRequestBuilder_region, "f");
        }
        data() {
            return {
                programs: this.programIds,
                season_id: this.seasonId,
                country: this.country,
                region: this.region,
                when: this.seasonId !== undefined ? When.PAST : undefined,
            };
        }
    }
});
export { TeamGroupsRequestBuilder };
export class TeamGroupsRequest extends TeamGroupsRequestBuilder.TeamGroupsRequest {
}
var When;
(function (When) {
    When["FUTURE"] = "future";
    When["PAST"] = "past";
})(When || (When = {}));
