// @robotevents/client/dist/v2/clients/rankings.js
var _EventDivisionRankingsRequestBuilder_eventId, _EventDivisionRankingsRequestBuilder_divisionId, _EventDivisionRankingsRequestBuilder_teamIds, _EventDivisionRankingsRequestBuilder_ranks, _TeamRankingsRequestBuilder_teamId, _TeamRankingsRequestBuilder_eventIds, _TeamRankingsRequestBuilder_ranks, _TeamRankingsRequestBuilder_seasonIds;
import { Client } from '../client.js';
import { PageableRequest, PageableRequestBuilder, } from '../pageable.js';
export class RankingsClient extends Client {
    findAllByEventDivision(request) {
        const req = request(new EventDivisionRankingsRequestBuilder()).build();
        return this.getAll(`/events/${req.eventId}/divisions/${req.divisionId}/rankings`, req.params());
    }
    findAllByTeam(request) {
        const req = request(new TeamRankingsRequestBuilder()).build();
        return this.getAll(`/teams/${req.teamId}/rankings`, req.params());
    }
    findAllFinalistsByEventDivision(request) {
        const req = request(new EventDivisionRankingsRequestBuilder()).build();
        return this.getAll(`/events/${req.eventId}/divisions/${req.divisionId}/finalistRankings`, req.params());
    }
}
class EventDivisionRankingsRequestBuilder extends PageableRequestBuilder {
    constructor() {
        super(...arguments);
        _EventDivisionRankingsRequestBuilder_eventId.set(this, void 0);
        _EventDivisionRankingsRequestBuilder_divisionId.set(this, void 0);
        _EventDivisionRankingsRequestBuilder_teamIds.set(this, void 0);
        _EventDivisionRankingsRequestBuilder_ranks.set(this, void 0);
    }
    eventId(value) {
        __classPrivateFieldSet(this, _EventDivisionRankingsRequestBuilder_eventId, value, "f");
        return this;
    }
    divisionId(value) {
        __classPrivateFieldSet(this, _EventDivisionRankingsRequestBuilder_divisionId, value, "f");
        return this;
    }
    teamIds(...value) {
        __classPrivateFieldSet(this, _EventDivisionRankingsRequestBuilder_teamIds, value, "f");
        return this;
    }
    ranks(...value) {
        __classPrivateFieldSet(this, _EventDivisionRankingsRequestBuilder_ranks, value, "f");
        return this;
    }
    build() {
        return new EventDivisionRankingsRequest(this);
    }
}
_EventDivisionRankingsRequestBuilder_eventId = new WeakMap(), _EventDivisionRankingsRequestBuilder_divisionId = new WeakMap(), _EventDivisionRankingsRequestBuilder_teamIds = new WeakMap(), _EventDivisionRankingsRequestBuilder_ranks = new WeakMap();
Object.defineProperty(EventDivisionRankingsRequestBuilder, "EventDivisionRankingsRequest", {
    enumerable: true,
    configurable: true,
    writable: true,
    value: class extends PageableRequest {
        constructor(builder) {
            super(builder);
            Object.defineProperty(this, "eventId", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "divisionId", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "teamIds", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "ranks", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            if (__classPrivateFieldGet(builder, _EventDivisionRankingsRequestBuilder_eventId, "f") === undefined) {
                throw new Error('Missing required property EventDivisionRankingsRequest.eventId');
            }
            if (__classPrivateFieldGet(builder, _EventDivisionRankingsRequestBuilder_divisionId, "f") === undefined) {
                throw new Error('Missing required property EventDivisionRankingsRequest.divisionId');
            }
            this.eventId = __classPrivateFieldGet(builder, _EventDivisionRankingsRequestBuilder_eventId, "f");
            this.divisionId = __classPrivateFieldGet(builder, _EventDivisionRankingsRequestBuilder_divisionId, "f");
            this.teamIds = __classPrivateFieldGet(builder, _EventDivisionRankingsRequestBuilder_teamIds, "f");
            this.ranks = __classPrivateFieldGet(builder, _EventDivisionRankingsRequestBuilder_ranks, "f");
        }
        params() {
            return { ...super.params(), team: this.teamIds, rank: this.ranks };
        }
    }
});
export { EventDivisionRankingsRequestBuilder };
export class EventDivisionRankingsRequest extends EventDivisionRankingsRequestBuilder.EventDivisionRankingsRequest {
}
class TeamRankingsRequestBuilder extends PageableRequestBuilder {
    constructor() {
        super(...arguments);
        _TeamRankingsRequestBuilder_teamId.set(this, void 0);
        _TeamRankingsRequestBuilder_eventIds.set(this, void 0);
        _TeamRankingsRequestBuilder_ranks.set(this, void 0);
        _TeamRankingsRequestBuilder_seasonIds.set(this, void 0);
    }
    teamId(value) {
        __classPrivateFieldSet(this, _TeamRankingsRequestBuilder_teamId, value, "f");
        return this;
    }
    eventIds(...value) {
        __classPrivateFieldSet(this, _TeamRankingsRequestBuilder_eventIds, value, "f");
        return this;
    }
    ranks(...value) {
        __classPrivateFieldSet(this, _TeamRankingsRequestBuilder_ranks, value, "f");
        return this;
    }
    seasonIds(...value) {
        __classPrivateFieldSet(this, _TeamRankingsRequestBuilder_seasonIds, value, "f");
        return this;
    }
    build() {
        return new TeamRankingsRequest(this);
    }
}
_TeamRankingsRequestBuilder_teamId = new WeakMap(), _TeamRankingsRequestBuilder_eventIds = new WeakMap(), _TeamRankingsRequestBuilder_ranks = new WeakMap(), _TeamRankingsRequestBuilder_seasonIds = new WeakMap();
Object.defineProperty(TeamRankingsRequestBuilder, "TeamRankingsRequest", {
    enumerable: true,
    configurable: true,
    writable: true,
    value: class extends PageableRequest {
        constructor(builder) {
            super(builder);
            Object.defineProperty(this, "teamId", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "eventIds", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "ranks", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "seasonIds", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            if (__classPrivateFieldGet(builder, _TeamRankingsRequestBuilder_teamId, "f") === undefined) {
                throw new Error('Missing required property TeamRankingsRequest.teamId');
            }
            this.teamId = __classPrivateFieldGet(builder, _TeamRankingsRequestBuilder_teamId, "f");
            this.eventIds = __classPrivateFieldGet(builder, _TeamRankingsRequestBuilder_eventIds, "f");
            this.ranks = __classPrivateFieldGet(builder, _TeamRankingsRequestBuilder_ranks, "f");
            this.seasonIds = __classPrivateFieldGet(builder, _TeamRankingsRequestBuilder_seasonIds, "f");
        }
        params() {
            return {
                ...super.params(),
                event: this.eventIds,
                rank: this.ranks,
                season: this.seasonIds,
            };
        }
    }
});
export { TeamRankingsRequestBuilder };
export class TeamRankingsRequest extends TeamRankingsRequestBuilder.TeamRankingsRequest {
}
