// @robotevents/client/dist/v2/clients/seasons.js
var _SeasonsRequestBuilder_ids, _SeasonsRequestBuilder_programIds, _SeasonsRequestBuilder_teamIds, _SeasonsRequestBuilder_start, _SeasonsRequestBuilder_end, _SeasonsRequestBuilder_active;
import { Client } from '../client.js';
import { PageableRequest, PageableRequestBuilder, } from '../pageable.js';
export class SeasonsClient extends Client {
    findAll(request) {
        return this.getAll('/seasons', request?.(new SeasonsRequestBuilder()).build().params());
    }
    async findById(id) {
        return this.get(`/seasons/${id}`);
    }
}
class SeasonsRequestBuilder extends PageableRequestBuilder {
    constructor() {
        super(...arguments);
        _SeasonsRequestBuilder_ids.set(this, void 0);
        _SeasonsRequestBuilder_programIds.set(this, void 0);
        _SeasonsRequestBuilder_teamIds.set(this, void 0);
        _SeasonsRequestBuilder_start.set(this, void 0);
        _SeasonsRequestBuilder_end.set(this, void 0);
        _SeasonsRequestBuilder_active.set(this, void 0);
    }
    ids(...value) {
        __classPrivateFieldSet(this, _SeasonsRequestBuilder_ids, value, "f");
        return this;
    }
    programIds(...value) {
        __classPrivateFieldSet(this, _SeasonsRequestBuilder_programIds, value, "f");
        return this;
    }
    teamIds(...value) {
        __classPrivateFieldSet(this, _SeasonsRequestBuilder_teamIds, value, "f");
        return this;
    }
    start(value) {
        __classPrivateFieldSet(this, _SeasonsRequestBuilder_start, value, "f");
        return this;
    }
    end(value) {
        __classPrivateFieldSet(this, _SeasonsRequestBuilder_end, value, "f");
        return this;
    }
    active(value) {
        __classPrivateFieldSet(this, _SeasonsRequestBuilder_active, value, "f");
        return this;
    }
    build() {
        return new SeasonsRequest(this);
    }
}
_SeasonsRequestBuilder_ids = new WeakMap(), _SeasonsRequestBuilder_programIds = new WeakMap(), _SeasonsRequestBuilder_teamIds = new WeakMap(), _SeasonsRequestBuilder_start = new WeakMap(), _SeasonsRequestBuilder_end = new WeakMap(), _SeasonsRequestBuilder_active = new WeakMap();
Object.defineProperty(SeasonsRequestBuilder, "SeasonsRequest", {
    enumerable: true,
    configurable: true,
    writable: true,
    value: class extends PageableRequest {
        constructor(builder) {
            super(builder);
            Object.defineProperty(this, "ids", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "programIds", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "teamIds", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "start", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "end", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "active", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            this.ids = __classPrivateFieldGet(builder, _SeasonsRequestBuilder_ids, "f");
            this.programIds = __classPrivateFieldGet(builder, _SeasonsRequestBuilder_programIds, "f");
            this.teamIds = __classPrivateFieldGet(builder, _SeasonsRequestBuilder_teamIds, "f");
            this.start = __classPrivateFieldGet(builder, _SeasonsRequestBuilder_start, "f");
            this.end = __classPrivateFieldGet(builder, _SeasonsRequestBuilder_end, "f");
            this.active = __classPrivateFieldGet(builder, _SeasonsRequestBuilder_active, "f");
        }
        params() {
            return {
                ...super.params(),
                id: this.ids,
                program: this.programIds,
                team: this.teamIds,
                start: this.start,
                end: this.end,
                active: this.active,
            };
        }
    }
});
export { SeasonsRequestBuilder };
export class SeasonsRequest extends SeasonsRequestBuilder.SeasonsRequest {
}
