// @robotevents/client/dist/v2/clients/teams.js
var _TeamsRequestBuilder_ids, _TeamsRequestBuilder_numbers, _TeamsRequestBuilder_eventIds, _TeamsRequestBuilder_registered, _TeamsRequestBuilder_programIds, _TeamsRequestBuilder_grades, _TeamsRequestBuilder_countries, _TeamsRequestBuilder_mine, _EventTeamsRequestBuilder_eventId, _EventTeamsRequestBuilder_numbers, _EventTeamsRequestBuilder_registered, _EventTeamsRequestBuilder_grades, _EventTeamsRequestBuilder_countries, _EventTeamsRequestBuilder_mine;
import { Client } from '../client.js';
import { PageableRequest, PageableRequestBuilder, } from '../pageable.js';
export class TeamsClient extends Client {
    findAll(request) {
        return this.getAll('/teams', request?.(new TeamsRequestBuilder()).build().params());
    }
    findAllByEvent(request) {
        const req = request(new EventTeamsRequestBuilder()).build();
        return this.getAll(`/events/${req.eventId}/teams`, req.params());
    }
    async findById(id) {
        return this.get(`/teams/${id}`);
    }
}
class TeamsRequestBuilder extends PageableRequestBuilder {
    constructor() {
        super(...arguments);
        _TeamsRequestBuilder_ids.set(this, void 0);
        _TeamsRequestBuilder_numbers.set(this, void 0);
        _TeamsRequestBuilder_eventIds.set(this, void 0);
        _TeamsRequestBuilder_registered.set(this, void 0);
        _TeamsRequestBuilder_programIds.set(this, void 0);
        _TeamsRequestBuilder_grades.set(this, void 0);
        _TeamsRequestBuilder_countries.set(this, void 0);
        _TeamsRequestBuilder_mine.set(this, void 0);
    }
    ids(...value) {
        __classPrivateFieldSet(this, _TeamsRequestBuilder_ids, value, "f");
        return this;
    }
    numbers(...value) {
        __classPrivateFieldSet(this, _TeamsRequestBuilder_numbers, value, "f");
        return this;
    }
    eventIds(...value) {
        __classPrivateFieldSet(this, _TeamsRequestBuilder_eventIds, value, "f");
        return this;
    }
    registered(value) {
        __classPrivateFieldSet(this, _TeamsRequestBuilder_registered, value, "f");
        return this;
    }
    programIds(...value) {
        __classPrivateFieldSet(this, _TeamsRequestBuilder_programIds, value, "f");
        return this;
    }
    grades(...value) {
        __classPrivateFieldSet(this, _TeamsRequestBuilder_grades, value, "f");
        return this;
    }
    countries(...value) {
        __classPrivateFieldSet(this, _TeamsRequestBuilder_countries, value, "f");
        return this;
    }
    mine(value) {
        __classPrivateFieldSet(this, _TeamsRequestBuilder_mine, value, "f");
        return this;
    }
    build() {
        return new TeamsRequest(this);
    }
}
_TeamsRequestBuilder_ids = new WeakMap(), _TeamsRequestBuilder_numbers = new WeakMap(), _TeamsRequestBuilder_eventIds = new WeakMap(), _TeamsRequestBuilder_registered = new WeakMap(), _TeamsRequestBuilder_programIds = new WeakMap(), _TeamsRequestBuilder_grades = new WeakMap(), _TeamsRequestBuilder_countries = new WeakMap(), _TeamsRequestBuilder_mine = new WeakMap();
Object.defineProperty(TeamsRequestBuilder, "TeamsRequest", {
    enumerable: true,
    configurable: true,
    writable: true,
    value: class extends PageableRequest {
        constructor(builder) {
            super(builder);
            Object.defineProperty(this, "ids", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "numbers", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "eventIds", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "registered", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "programIds", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "grades", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "countries", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "mine", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            this.ids = __classPrivateFieldGet(builder, _TeamsRequestBuilder_ids, "f");
            this.numbers = __classPrivateFieldGet(builder, _TeamsRequestBuilder_numbers, "f");
            this.eventIds = __classPrivateFieldGet(builder, _TeamsRequestBuilder_eventIds, "f");
            this.registered = __classPrivateFieldGet(builder, _TeamsRequestBuilder_registered, "f");
            this.programIds = __classPrivateFieldGet(builder, _TeamsRequestBuilder_programIds, "f");
            this.grades = __classPrivateFieldGet(builder, _TeamsRequestBuilder_grades, "f");
            this.countries = __classPrivateFieldGet(builder, _TeamsRequestBuilder_countries, "f");
            this.mine = __classPrivateFieldGet(builder, _TeamsRequestBuilder_mine, "f");
        }
        params() {
            return {
                ...super.params(),
                id: this.ids,
                number: this.numbers,
                event: this.eventIds,
                registered: this.registered,
                program: this.programIds,
                grade: this.grades,
                country: this.countries,
                myTeams: this.mine,
            };
        }
    }
});
export { TeamsRequestBuilder };
export class TeamsRequest extends TeamsRequestBuilder.TeamsRequest {
}
class EventTeamsRequestBuilder extends PageableRequestBuilder {
    constructor() {
        super(...arguments);
        _EventTeamsRequestBuilder_eventId.set(this, void 0);
        _EventTeamsRequestBuilder_numbers.set(this, void 0);
        _EventTeamsRequestBuilder_registered.set(this, void 0);
        _EventTeamsRequestBuilder_grades.set(this, void 0);
        _EventTeamsRequestBuilder_countries.set(this, void 0);
        _EventTeamsRequestBuilder_mine.set(this, void 0);
    }
    eventId(value) {
        __classPrivateFieldSet(this, _EventTeamsRequestBuilder_eventId, value, "f");
        return this;
    }
    numbers(...value) {
        __classPrivateFieldSet(this, _EventTeamsRequestBuilder_numbers, value, "f");
        return this;
    }
    registered(value) {
        __classPrivateFieldSet(this, _EventTeamsRequestBuilder_registered, value, "f");
        return this;
    }
    grades(...value) {
        __classPrivateFieldSet(this, _EventTeamsRequestBuilder_grades, value, "f");
        return this;
    }
    countries(...value) {
        __classPrivateFieldSet(this, _EventTeamsRequestBuilder_countries, value, "f");
        return this;
    }
    mine(value) {
        __classPrivateFieldSet(this, _EventTeamsRequestBuilder_mine, value, "f");
        return this;
    }
    build() {
        return new EventTeamsRequest(this);
    }
}
_EventTeamsRequestBuilder_eventId = new WeakMap(), _EventTeamsRequestBuilder_numbers = new WeakMap(), _EventTeamsRequestBuilder_registered = new WeakMap(), _EventTeamsRequestBuilder_grades = new WeakMap(), _EventTeamsRequestBuilder_countries = new WeakMap(), _EventTeamsRequestBuilder_mine = new WeakMap();
Object.defineProperty(EventTeamsRequestBuilder, "EventTeamsRequest", {
    enumerable: true,
    configurable: true,
    writable: true,
    value: class extends PageableRequest {
        constructor(builder) {
            super(builder);
            Object.defineProperty(this, "eventId", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "numbers", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "registered", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "grades", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "countries", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "mine", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            if (__classPrivateFieldGet(builder, _EventTeamsRequestBuilder_eventId, "f") === undefined) {
                throw new Error('Missing required property EventTeamsRequest.eventId');
            }
            this.eventId = __classPrivateFieldGet(builder, _EventTeamsRequestBuilder_eventId, "f");
            this.numbers = __classPrivateFieldGet(builder, _EventTeamsRequestBuilder_numbers, "f");
            this.registered = __classPrivateFieldGet(builder, _EventTeamsRequestBuilder_registered, "f");
            this.grades = __classPrivateFieldGet(builder, _EventTeamsRequestBuilder_grades, "f");
            this.countries = __classPrivateFieldGet(builder, _EventTeamsRequestBuilder_countries, "f");
            this.mine = __classPrivateFieldGet(builder, _EventTeamsRequestBuilder_mine, "f");
        }
        params() {
            return {
                ...super.params(),
                number: this.numbers,
                registered: this.registered,
                grade: this.grades,
                country: this.countries,
                myTeams: this.mine,
            };
        }
    }
});
export { EventTeamsRequestBuilder };
export class EventTeamsRequest extends EventTeamsRequestBuilder.EventTeamsRequest {
}
