// @robotevents/client/dist/v2/clients/events.js
var _EventsRequestBuilder_ids, _EventsRequestBuilder_skus, _EventsRequestBuilder_teamIds, _EventsRequestBuilder_seasonIds, _EventsRequestBuilder_start, _EventsRequestBuilder_end, _EventsRequestBuilder_levels, _EventsRequestBuilder_mine, _EventsRequestBuilder_types, _SeasonEventsRequestBuilder_seasonId, _SeasonEventsRequestBuilder_skus, _SeasonEventsRequestBuilder_teamIds, _SeasonEventsRequestBuilder_start, _SeasonEventsRequestBuilder_end, _SeasonEventsRequestBuilder_levels, _TeamEventsRequestBuilder_teamId, _TeamEventsRequestBuilder_skus, _TeamEventsRequestBuilder_seasonIds, _TeamEventsRequestBuilder_start, _TeamEventsRequestBuilder_end, _TeamEventsRequestBuilder_levels;
import { Client } from '../client.js';
import { PageableRequest, PageableRequestBuilder, } from '../pageable.js';
export class EventsClient extends Client {
    findAll(request) {
        return this.getAll('/events', request?.(new EventsRequestBuilder()).build().params());
    }
    findAllBySeason(request) {
        const req = request(new SeasonEventsRequestBuilder()).build();
        return this.getAll(`/seasons/${req.seasonId}/events`, req.params());
    }
    findAllByTeam(request) {
        const req = request(new TeamEventsRequestBuilder()).build();
        return this.getAll(`/teams/${req.teamId}/events`, req.params());
    }
    async findById(id) {
        return this.get(`/events/${id}`);
    }
}
class EventsRequestBuilder extends PageableRequestBuilder {
    constructor() {
        super(...arguments);
        _EventsRequestBuilder_ids.set(this, void 0);
        _EventsRequestBuilder_skus.set(this, void 0);
        _EventsRequestBuilder_teamIds.set(this, void 0);
        _EventsRequestBuilder_seasonIds.set(this, void 0);
        _EventsRequestBuilder_start.set(this, void 0);
        _EventsRequestBuilder_end.set(this, void 0);
        _EventsRequestBuilder_levels.set(this, void 0);
        _EventsRequestBuilder_mine.set(this, void 0);
        _EventsRequestBuilder_types.set(this, void 0);
    }
    ids(...value) {
        __classPrivateFieldSet(this, _EventsRequestBuilder_ids, value, "f");
        return this;
    }
    skus(...value) {
        __classPrivateFieldSet(this, _EventsRequestBuilder_skus, value, "f");
        return this;
    }
    teamIds(...value) {
        __classPrivateFieldSet(this, _EventsRequestBuilder_teamIds, value, "f");
        return this;
    }
    seasonIds(...value) {
        __classPrivateFieldSet(this, _EventsRequestBuilder_seasonIds, value, "f");
        return this;
    }
    start(value) {
        __classPrivateFieldSet(this, _EventsRequestBuilder_start, value, "f");
        return this;
    }
    end(value) {
        __classPrivateFieldSet(this, _EventsRequestBuilder_end, value, "f");
        return this;
    }
    levels(...value) {
        __classPrivateFieldSet(this, _EventsRequestBuilder_levels, value, "f");
        return this;
    }
    mine(value) {
        __classPrivateFieldSet(this, _EventsRequestBuilder_mine, value, "f");
        return this;
    }
    types(...value) {
        __classPrivateFieldSet(this, _EventsRequestBuilder_types, value, "f");
        return this;
    }
    build() {
        return new EventsRequest(this);
    }
}
_EventsRequestBuilder_ids = new WeakMap(), _EventsRequestBuilder_skus = new WeakMap(), _EventsRequestBuilder_teamIds = new WeakMap(), _EventsRequestBuilder_seasonIds = new WeakMap(), _EventsRequestBuilder_start = new WeakMap(), _EventsRequestBuilder_end = new WeakMap(), _EventsRequestBuilder_levels = new WeakMap(), _EventsRequestBuilder_mine = new WeakMap(), _EventsRequestBuilder_types = new WeakMap();
Object.defineProperty(EventsRequestBuilder, "EventsRequest", {
    enumerable: true,
    configurable: true,
    writable: true,
    value: class extends PageableRequest {
        constructor(builder) {
            super(builder);
            Object.defineProperty(this, "ids", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "skus", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "teamIds", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "seasonIds", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "start", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "end", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "levels", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "mine", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "types", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            this.ids = __classPrivateFieldGet(builder, _EventsRequestBuilder_ids, "f");
            this.skus = __classPrivateFieldGet(builder, _EventsRequestBuilder_skus, "f");
            this.teamIds = __classPrivateFieldGet(builder, _EventsRequestBuilder_teamIds, "f");
            this.seasonIds = __classPrivateFieldGet(builder, _EventsRequestBuilder_seasonIds, "f");
            this.start = __classPrivateFieldGet(builder, _EventsRequestBuilder_start, "f");
            this.end = __classPrivateFieldGet(builder, _EventsRequestBuilder_end, "f");
            this.levels = __classPrivateFieldGet(builder, _EventsRequestBuilder_levels, "f");
            this.mine = __classPrivateFieldGet(builder, _EventsRequestBuilder_mine, "f");
            this.types = __classPrivateFieldGet(builder, _EventsRequestBuilder_types, "f");
        }
        params() {
            return {
                ...super.params(),
                id: this.ids,
                sku: this.skus,
                team: this.teamIds,
                season: this.seasonIds,
                start: this.start,
                end: this.end,
                level: this.levels,
                myEvents: this.mine,
                eventType: this.types,
            };
        }
    }
});
export { EventsRequestBuilder };
export class EventsRequest extends EventsRequestBuilder.EventsRequest {
}
class SeasonEventsRequestBuilder extends PageableRequestBuilder {
    constructor() {
        super(...arguments);
        _SeasonEventsRequestBuilder_seasonId.set(this, void 0);
        _SeasonEventsRequestBuilder_skus.set(this, void 0);
        _SeasonEventsRequestBuilder_teamIds.set(this, void 0);
        _SeasonEventsRequestBuilder_start.set(this, void 0);
        _SeasonEventsRequestBuilder_end.set(this, void 0);
        _SeasonEventsRequestBuilder_levels.set(this, void 0);
    }
    seasonId(value) {
        __classPrivateFieldSet(this, _SeasonEventsRequestBuilder_seasonId, value, "f");
        return this;
    }
    skus(...value) {
        __classPrivateFieldSet(this, _SeasonEventsRequestBuilder_skus, value, "f");
        return this;
    }
    teamIds(...value) {
        __classPrivateFieldSet(this, _SeasonEventsRequestBuilder_teamIds, value, "f");
        return this;
    }
    start(value) {
        __classPrivateFieldSet(this, _SeasonEventsRequestBuilder_start, value, "f");
        return this;
    }
    end(value) {
        __classPrivateFieldSet(this, _SeasonEventsRequestBuilder_end, value, "f");
        return this;
    }
    levels(...value) {
        __classPrivateFieldSet(this, _SeasonEventsRequestBuilder_levels, value, "f");
        return this;
    }
    build() {
        return new SeasonEventsRequest(this);
    }
}
_SeasonEventsRequestBuilder_seasonId = new WeakMap(), _SeasonEventsRequestBuilder_skus = new WeakMap(), _SeasonEventsRequestBuilder_teamIds = new WeakMap(), _SeasonEventsRequestBuilder_start = new WeakMap(), _SeasonEventsRequestBuilder_end = new WeakMap(), _SeasonEventsRequestBuilder_levels = new WeakMap();
Object.defineProperty(SeasonEventsRequestBuilder, "SeasonEventsRequest", {
    enumerable: true,
    configurable: true,
    writable: true,
    value: class extends PageableRequest {
        constructor(builder) {
            super(builder);
            Object.defineProperty(this, "seasonId", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "skus", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "teamIds", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "start", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "end", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "levels", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            if (__classPrivateFieldGet(builder, _SeasonEventsRequestBuilder_seasonId, "f") === undefined) {
                throw new Error('Missing required property SeasonEventsRequest.seasonId');
            }
            this.seasonId = __classPrivateFieldGet(builder, _SeasonEventsRequestBuilder_seasonId, "f");
            this.skus = __classPrivateFieldGet(builder, _SeasonEventsRequestBuilder_skus, "f");
            this.teamIds = __classPrivateFieldGet(builder, _SeasonEventsRequestBuilder_teamIds, "f");
            this.start = __classPrivateFieldGet(builder, _SeasonEventsRequestBuilder_start, "f");
            this.end = __classPrivateFieldGet(builder, _SeasonEventsRequestBuilder_end, "f");
            this.levels = __classPrivateFieldGet(builder, _SeasonEventsRequestBuilder_levels, "f");
        }
        params() {
            return {
                ...super.params(),
                sku: this.skus,
                team: this.teamIds,
                start: this.start,
                end: this.end,
                level: this.levels,
            };
        }
    }
});
export { SeasonEventsRequestBuilder };
export class SeasonEventsRequest extends SeasonEventsRequestBuilder.SeasonEventsRequest {
}
class TeamEventsRequestBuilder extends PageableRequestBuilder {
    constructor() {
        super(...arguments);
        _TeamEventsRequestBuilder_teamId.set(this, void 0);
        _TeamEventsRequestBuilder_skus.set(this, void 0);
        _TeamEventsRequestBuilder_seasonIds.set(this, void 0);
        _TeamEventsRequestBuilder_start.set(this, void 0);
        _TeamEventsRequestBuilder_end.set(this, void 0);
        _TeamEventsRequestBuilder_levels.set(this, void 0);
    }
    teamId(value) {
        __classPrivateFieldSet(this, _TeamEventsRequestBuilder_teamId, value, "f");
        return this;
    }
    skus(...value) {
        __classPrivateFieldSet(this, _TeamEventsRequestBuilder_skus, value, "f");
        return this;
    }
    seasonIds(...value) {
        __classPrivateFieldSet(this, _TeamEventsRequestBuilder_seasonIds, value, "f");
        return this;
    }
    start(value) {
        __classPrivateFieldSet(this, _TeamEventsRequestBuilder_start, value, "f");
        return this;
    }
    end(value) {
        __classPrivateFieldSet(this, _TeamEventsRequestBuilder_end, value, "f");
        return this;
    }
    levels(...value) {
        __classPrivateFieldSet(this, _TeamEventsRequestBuilder_levels, value, "f");
        return this;
    }
    build() {
        return new TeamEventsRequest(this);
    }
}
_TeamEventsRequestBuilder_teamId = new WeakMap(), _TeamEventsRequestBuilder_skus = new WeakMap(), _TeamEventsRequestBuilder_seasonIds = new WeakMap(), _TeamEventsRequestBuilder_start = new WeakMap(), _TeamEventsRequestBuilder_end = new WeakMap(), _TeamEventsRequestBuilder_levels = new WeakMap();
Object.defineProperty(TeamEventsRequestBuilder, "TeamEventsRequest", {
    enumerable: true,
    configurable: true,
    writable: true,
    value: class extends PageableRequest {
        constructor(builder) {
            super(builder);
            Object.defineProperty(this, "teamId", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "skus", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "seasonIds", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "start", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "end", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "levels", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            if (__classPrivateFieldGet(builder, _TeamEventsRequestBuilder_teamId, "f") === undefined) {
                throw new Error('Missing required property TeamEventsRequest.teamId');
            }
            this.teamId = __classPrivateFieldGet(builder, _TeamEventsRequestBuilder_teamId, "f");
            this.skus = __classPrivateFieldGet(builder, _TeamEventsRequestBuilder_skus, "f");
            this.seasonIds = __classPrivateFieldGet(builder, _TeamEventsRequestBuilder_seasonIds, "f");
            this.start = __classPrivateFieldGet(builder, _TeamEventsRequestBuilder_start, "f");
            this.end = __classPrivateFieldGet(builder, _TeamEventsRequestBuilder_end, "f");
            this.levels = __classPrivateFieldGet(builder, _TeamEventsRequestBuilder_levels, "f");
        }
        params() {
            return {
                ...super.params(),
                sku: this.skus,
                season: this.seasonIds,
                start: this.start,
                end: this.end,
                level: this.levels,
            };
        }
    }
});
export { TeamEventsRequestBuilder };
export class TeamEventsRequest extends TeamEventsRequestBuilder.TeamEventsRequest {
}
