// @robotevents/client/dist/v2/clients/index.js
export * from './awards.js';
export * from './events.js';
export * from './matches.js';
export * from './programs.js';
export * from './rankings.js';
export * from './seasons.js';
export * from './skills.js';
export * from './teams.js';
export var AllianceColor;
(function (AllianceColor) {
    AllianceColor["Red"] = "red";
    AllianceColor["Blue"] = "blue";
})(AllianceColor || (AllianceColor = {}));
export var AwardClassification;
(function (AwardClassification) {
    AwardClassification["Champion"] = "champion";
    AwardClassification["Finalist"] = "finalist";
    AwardClassification["Semifinalist"] = "semifinalist";
    AwardClassification["Quarterfinalist"] = "quarterfinalist";
})(AwardClassification || (AwardClassification = {}));
export var AwardDesignation;
(function (AwardDesignation) {
    AwardDesignation["Tournament"] = "tournament";
    AwardDesignation["Division"] = "division";
})(AwardDesignation || (AwardDesignation = {}));
export var EventLevel;
(function (EventLevel) {
    EventLevel["World"] = "World";
    EventLevel["National"] = "National";
    EventLevel["Regional"] = "Regional";
    EventLevel["State"] = "State";
    EventLevel["Signature"] = "Signature";
    EventLevel["Other"] = "Other";
})(EventLevel || (EventLevel = {}));
export var EventType;
(function (EventType) {
    EventType["Tournament"] = "tournament";
    EventType["League"] = "league";
    EventType["Workshop"] = "workshop";
    EventType["Virtual"] = "virtual";
})(EventType || (EventType = {}));
export var Grade;
(function (Grade) {
    Grade["College"] = "College";
    Grade["HighSchool"] = "High School";
    Grade["MiddleSchool"] = "Middle School";
    Grade["ElementarySchool"] = "Elementary School";
})(Grade || (Grade = {}));
export var ProgramId;
(function (ProgramId) {
    ProgramId[ProgramId["VRC"] = 1] = "VRC";
    ProgramId[ProgramId["VEXU"] = 4] = "VEXU";
    ProgramId[ProgramId["Workshop"] = 37] = "Workshop";
    ProgramId[ProgramId["VIQC"] = 41] = "VIQC";
    ProgramId[ProgramId["NRL"] = 43] = "NRL";
    ProgramId[ProgramId["ADC"] = 44] = "ADC";
    ProgramId[ProgramId["TVRC"] = 46] = "TVRC";
    ProgramId[ProgramId["TIQC"] = 47] = "TIQC";
    ProgramId[ProgramId["VRAD"] = 51] = "VRAD";
    ProgramId[ProgramId["BellAVR"] = 55] = "BellAVR";
    ProgramId[ProgramId["FAC"] = 56] = "FAC";
    ProgramId[ProgramId["VAIC"] = 57] = "VAIC";
})(ProgramId || (ProgramId = {}));
export var Round;
(function (Round) {
    Round[Round["Practice"] = 1] = "Practice";
    Round[Round["Qualification"] = 2] = "Qualification";
    Round[Round["Quarterfinal"] = 3] = "Quarterfinal";
    Round[Round["Semifinal"] = 4] = "Semifinal";
    Round[Round["Final"] = 5] = "Final";
    Round[Round["RoundOf16"] = 6] = "RoundOf16";
    Round[Round["RoundOf32"] = 7] = "RoundOf32";
    Round[Round["RoundOf64"] = 8] = "RoundOf64";
    Round[Round["RoundOf128"] = 9] = "RoundOf128";
    Round[Round["TopN"] = 15] = "TopN";
    Round[Round["RoundRobin"] = 16] = "RoundRobin";
})(Round || (Round = {}));
export var SkillType;
(function (SkillType) {
    SkillType["Driver"] = "driver";
    SkillType["Programming"] = "programming";
})(SkillType || (SkillType = {}));
