// @robotevents/client/dist/v2/clients/programs.js
var _ProgramsRequestBuilder_ids;
import { Client } from '../client.js';
import { PageableRequest, PageableRequestBuilder, } from '../pageable.js';
export class ProgramsClient extends Client {
    findAll(request) {
        return this.getAll('/programs', request?.(new ProgramsRequestBuilder()).build().params());
    }
    async findById(id) {
        return this.get(`/programs/${id}`);
    }
}
class ProgramsRequestBuilder extends PageableRequestBuilder {
    constructor() {
        super(...arguments);
        _ProgramsRequestBuilder_ids.set(this, void 0);
    }
    ids(...value) {
        __classPrivateFieldSet(this, _ProgramsRequestBuilder_ids, value, "f");
        return this;
    }
    build() {
        return new ProgramsRequest(this);
    }
}
_ProgramsRequestBuilder_ids = new WeakMap();
Object.defineProperty(ProgramsRequestBuilder, "ProgramsRequest", {
    enumerable: true,
    configurable: true,
    writable: true,
    value: class extends PageableRequest {
        constructor(builder) {
            super(builder);
            Object.defineProperty(this, "ids", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            this.ids = __classPrivateFieldGet(builder, _ProgramsRequestBuilder_ids, "f");
        }
        params() {
            return { ...super.params(), id: this.ids };
        }
    }
});
export { ProgramsRequestBuilder };
export class ProgramsRequest extends ProgramsRequestBuilder.ProgramsRequest {
}
