// @robotevents/client/dist/v1/clients/events.js
var _EventsRequestBuilder_programIds, _EventsRequestBuilder_seasonId, _EventsRequestBuilder_countryId, _EventsRequestBuilder_region;
import { Client } from '../client.js';
export class EventsClient extends Client {
    async findAll(request) {
        return await this.post('/events', request(new EventsRequestBuilder()).build().data());
    }
}
class EventsRequestBuilder {
    constructor() {
        _EventsRequestBuilder_programIds.set(this, void 0);
        _EventsRequestBuilder_seasonId.set(this, void 0);
        _EventsRequestBuilder_countryId.set(this, void 0);
        _EventsRequestBuilder_region.set(this, void 0);
    }
    programIds(...value) {
        __classPrivateFieldSet(this, _EventsRequestBuilder_programIds, value, "f");
        return this;
    }
    seasonId(value) {
        __classPrivateFieldSet(this, _EventsRequestBuilder_seasonId, value, "f");
        return this;
    }
    location(countryId, region) {
        __classPrivateFieldSet(this, _EventsRequestBuilder_countryId, countryId, "f");
        __classPrivateFieldSet(this, _EventsRequestBuilder_region, region, "f");
        return this;
    }
    build() {
        return new EventsRequest(this);
    }
}
_EventsRequestBuilder_programIds = new WeakMap(), _EventsRequestBuilder_seasonId = new WeakMap(), _EventsRequestBuilder_countryId = new WeakMap(), _EventsRequestBuilder_region = new WeakMap();
Object.defineProperty(EventsRequestBuilder, "EventsRequest", {
    enumerable: true,
    configurable: true,
    writable: true,
    value: class EventsRequest {
        constructor(builder) {
            Object.defineProperty(this, "programIds", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "seasonId", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "countryId", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "region", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            if (!__classPrivateFieldGet(builder, _EventsRequestBuilder_programIds, "f")?.length) {
                throw new Error('Missing required property EventsRequest.programIds');
            }
            this.programIds = __classPrivateFieldGet(builder, _EventsRequestBuilder_programIds, "f");
            this.seasonId = __classPrivateFieldGet(builder, _EventsRequestBuilder_seasonId, "f");
            this.countryId = __classPrivateFieldGet(builder, _EventsRequestBuilder_countryId, "f");
            this.region = __classPrivateFieldGet(builder, _EventsRequestBuilder_region, "f");
        }
        data() {
            return {
                programs: this.programIds,
                season_id: this.seasonId,
                country: this.countryId,
                region: this.region,
                when: this.seasonId !== undefined ? When.PAST : undefined,
            };
        }
    }
});
export { EventsRequestBuilder };
export class EventsRequest extends EventsRequestBuilder.EventsRequest {
}
var When;
(function (When) {
    When["FUTURE"] = "future";
    When["PAST"] = "past";
})(When || (When = {}));
