// @robotevents/client/dist/v1/clients/skills.js
var _SeasonSkillsRequestBuilder_seasonId, _SeasonSkillsRequestBuilder_postseason, _SeasonSkillsRequestBuilder_grade;
import axios from 'axios';
import { Client } from '../client.js';
export class SkillsClient extends Client {
    async findAllBySeason(request) {
        const req = request(new SeasonSkillsRequestBuilder()).build();
        try {
            return await this.get(`/seasons/${req.seasonId}/skills`, req.params());
        }
        catch (error) {
            if (axios.isAxiosError(error) && error.response?.status === 404) {
                return [];
            }
            throw error;
        }
    }
}
class SeasonSkillsRequestBuilder {
    constructor() {
        _SeasonSkillsRequestBuilder_seasonId.set(this, void 0);
        _SeasonSkillsRequestBuilder_postseason.set(this, void 0);
        _SeasonSkillsRequestBuilder_grade.set(this, void 0);
    }
    seasonId(value) {
        __classPrivateFieldSet(this, _SeasonSkillsRequestBuilder_seasonId, value, "f");
        return this;
    }
    postseason(value) {
        __classPrivateFieldSet(this, _SeasonSkillsRequestBuilder_postseason, value, "f");
        return this;
    }
    grade(value) {
        __classPrivateFieldSet(this, _SeasonSkillsRequestBuilder_grade, value, "f");
        return this;
    }
    build() {
        return new SeasonSkillsRequest(this);
    }
}
_SeasonSkillsRequestBuilder_seasonId = new WeakMap(), _SeasonSkillsRequestBuilder_postseason = new WeakMap(), _SeasonSkillsRequestBuilder_grade = new WeakMap();
Object.defineProperty(SeasonSkillsRequestBuilder, "SeasonSkillsRequest", {
    enumerable: true,
    configurable: true,
    writable: true,
    value: class {
        constructor(builder) {
            Object.defineProperty(this, "seasonId", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "postseason", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            Object.defineProperty(this, "grade", {
                enumerable: true,
                configurable: true,
                writable: true,
                value: void 0
            });
            if (__classPrivateFieldGet(builder, _SeasonSkillsRequestBuilder_seasonId, "f") === undefined) {
                throw new Error('Missing required property SeasonSkillsRequest.seasonId');
            }
            this.seasonId = __classPrivateFieldGet(builder, _SeasonSkillsRequestBuilder_seasonId, "f");
            this.postseason = __classPrivateFieldGet(builder, _SeasonSkillsRequestBuilder_postseason, "f");
            this.grade = __classPrivateFieldGet(builder, _SeasonSkillsRequestBuilder_grade, "f");
        }
        params() {
            return {
                post_season: this.postseason,
                grade_level: this.grade,
            };
        }
    }
});
export { SeasonSkillsRequestBuilder };
export class SeasonSkillsRequest extends SeasonSkillsRequestBuilder.SeasonSkillsRequest {
}
